### PR TITLE
[fix](docker case) Fix `test_sql_mode_node_mgr` and add cloud multi f…

### PR DESCRIPTION
**Pull request by deardeng**
…ollower fuzzy test

### What problem does this PR solve?
1. fix start docker cluster slow, double cycle use one break, can't break out
2. Fix the test_sql_mode_node_mgr case that fails to meet testing objectives, as the container exits after dropping the follower.
3. Add fuzzy add/drop follower/observer cases on the cloud.
4. Fix docker compose error `java.lang.Exception: doris compose output is empty, err:`
5. Fix show frontends slow query when change fe nodes, due to `show frontends` use is isJoin function call getHostName(https://bugs.openjdk.org/browse/JDK-8143378#:~:text=getHostName()%3B%20takes%20about%205,millisecond%20on%20JDK%20update%2051.)
6. and Fix 
```
2024-11-28 15:50:05,272 WARN (cloud cluster check|302) [CloudClusterChecker.checkCloudBackends():518] diff cluster has exception, Cannot invoke "org.apache.doris.system.Backend.isDecommissioned()" because "be" is null
java.lang.NullPointerException: Cannot invoke "org.apache.doris.system.Backend.isDecommissioned()" because "be" is null
        at org.apache.doris.cloud.catalog.CloudClusterChecker.updateStatus(CloudClusterChecker.java:175) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.cloud.catalog.CloudClusterChecker.checkDiffNode(CloudClusterChecker.java:263) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.cloud.catalog.CloudClusterChecker.checkCloudBackends(CloudClusterChecker.java:513) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.cloud.catalog.CloudClusterChecker.runAfterCatalogReady(CloudClusterChecker.java:320) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.common.util.MasterDaemon.runOneCycle(MasterDaemon.java:58) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.common.util.Daemon.run(Daemon.java:119) ~[doris-fe.jar:1.2-SNAPSHOT]
```

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Review by yujun777 — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/common/proc/FrontendsProcNode.java [hunk 1]
```diff
@@ -227,6 +222,15 @@ private static boolean isJoin(List<InetSocketAddress> allFeHosts, Frontend fe) {
                 return true;
             }
         }
+
+        for (InetSocketAddress addr : allFeHosts) {
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/common/proc/FrontendsProcNode.java:226]: check editog port

Comment (deardeng) [reply fe/fe-core/src/main/java/org/apache/doris/common/proc/FrontendsProcNode.java:227]: fixed

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Review by yujun777 — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Review by yujun777 — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**